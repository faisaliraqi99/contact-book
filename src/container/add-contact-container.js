import React, { Component } from 'react';
import { connect } from 'react-redux';

import AddContact from '../component/AddContact'
import { createContact } from '../actions/index';

export class AddContactContainer extends Component {
  state = {
    data: {
      "id": 0,
      "name": "Default",
      "lastname": "Default",
      "number": "0550112233",
      "countrycode": "+996",
      "email": "undefined@gmail.com",
      "address": "Undefined"
    }
  }

  componentDidUpdate = (prevProps) => {
    const contactsArrProps = this.props.state.contacts;

    if (contactsArrProps.length !== prevProps.state.contacts.length) {
      const index = contactsArrProps.length - 1;
      this.setState({ data: { ...this.state.data, id: contactsArrProps[index].id + 1} });
    }
  }

  createContact = () => {
    this.props.createContacts(this.state.data)
      .then(() => alert(`Контакт ${this.state.data.name} добавлен`))
      .catch(error => console.log(error));
  }

  editState = (event) => {
    const inputId = event.target.id;
    const inputVal = event.target.value;

    this.setState({ data: { ...this.state.data, [inputId]: inputVal } });
  }

  render() {
    return (
      <AddContact
        createContact={this.createContact}
        editState={this.editState}
      />
    );
  }
}

const mapStateToProps = (state) => {
  return state
};

export const mapDispatchToProps = dispatch => {
  return {
    createContacts: (data) => dispatch(createContact(data))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AddContactContainer);