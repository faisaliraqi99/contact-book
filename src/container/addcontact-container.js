import React, { Component } from 'react';

class AddContactContainer extends Component {
    render() {
        return (
            <div>
                Hello world!
            </div>
        );
    }
}

export default AddContactContainer;