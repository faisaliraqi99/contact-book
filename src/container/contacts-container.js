import React, { Component } from 'react';
import { connect } from 'react-redux';

import { addSelectedContact } from '../actions/index';
import Contacts from '../component/Contacts';

export class ContactsContainer extends Component {
  selectItem = (event) => {
    const contactsList = this.props.state.contacts;
    const contactId = event.target.closest('li').getAttribute('id');
    const index = contactsList.map((o) => o.id).indexOf(+contactId);

    const data = contactsList[index];
    this.props.addSelectedContacts(data);
  }

  render() {
    return (
      <Contacts
        contacts={this.props.state.contacts}
        selectItem={this.selectItem}
      />
    )
  }
}

export const mapStateToProps = (state) => {
  return state
};

export const mapDispatchToProps = dispatch => {
  return {
    addSelectedContacts: (data) => dispatch(addSelectedContact(data))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ContactsContainer);