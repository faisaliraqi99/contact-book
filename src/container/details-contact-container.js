import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

// import * as actionCreators from '../actions/index.js';
import { deleteContact } from '../actions/index';
import DetailsContact from '../component/DetailsContact'

export class DetailsContactContainer extends Component {
  editContact = () => {
    this.props.history.push('/edit');
  }
  deleteContact = (id) => {
    if (window.confirm('Вы действительно хотите удалить контакт?')) {
      this.props.deleteContacts(id)
        .then(this.props.history.push('/'))
        .catch(error => alert(error));
    }
  }
  render() {
    return (
      <DetailsContact
        contactData={this.props.state.selectedContact}
        deleteContact={this.deleteContact}
        editContact={this.editContact}
      />
    );
  }
}

export const mapStateToProps = (state) => {
  return state
};

export const mapDispatchToProps = dispatch => {
  return {
    deleteContacts: (data) => dispatch(deleteContact(data))
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(DetailsContactContainer));