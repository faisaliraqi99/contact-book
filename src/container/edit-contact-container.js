import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actionCreators from '../actions/index.js';
import { withRouter } from 'react-router-dom';

import EditContact from '../component/EditContact'
import { actionEditContact } from '../actions/index';

class EditContactContainer extends Component {
    state = {
        "id": `${this.props.state.selectedContact.id}`,
        "name": `${this.props.state.selectedContact.name}`,
        "lastname": `${this.props.state.selectedContact.lastname}`,
        "number": `${this.props.state.selectedContact.number}`,
        "countrycode": `${this.props.state.selectedContact.countrycode}`,
        "email": `${this.props.state.selectedContact.email}`,
        "address": `${this.props.state.selectedContact.address}`
    }

    editState = (event) => {
        const inputId = event.target.id;
        const inputVal = event.target.value;

        this.setState({ [inputId]: inputVal});
    }
    saveContact = () => {
        this.props.store.dispatch(actionEditContact(this.state.id, this.state))
            .then(this.props.history.push('/'))
            .catch(err => {
                console.log(err);
            });
    }
    render() {
        return (
            <EditContact
                contactData={this.props.state.selectedContact}
                editState={this.editState}
                saveContact={this.saveContact}
            />
        );
    }
}

const mapStateToProps = (state) => {
    return state
};

export default withRouter(connect(mapStateToProps, actionCreators)(EditContactContainer));