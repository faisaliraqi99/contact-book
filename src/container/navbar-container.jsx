import React, { Component } from 'react';
import NavBar from '../component/NavBar';

class NavBarContainer extends Component {
    render() {
        return (
            <NavBar></NavBar>
        );
    }
}

export default NavBarContainer;