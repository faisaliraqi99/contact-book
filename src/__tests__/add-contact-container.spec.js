import React from 'react'
import { mount, shallow } from 'enzyme'
import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import rootReducer from '../reducers'

import { mapDispatchToProps } from '../container/add-contact-container'
import AddContactContainer, { AddContactContainer as OriginalAddContactContainer } from '../container/add-contact-container'

const store = createStore(rootReducer, applyMiddleware(thunk));

const initialData = {
  contacts: [
    {
      id: 1,
      name: 'Somename',
      lastname: 'Lastname',
      number: '550121212',
      countrycode: '+996',
      email: 'undefined@mail.ru',
      address: 'Someaddr'
    },
    {
      id: 2,
      name: 'Somename',
      lastname: 'Lastname',
      number: '550121212',
      countrycode: '+996',
      email: 'undefined@mail.ru',
      address: 'Someaddr'
    }
  ]
}

describe('AddContactContainer test group', () => {
  const props = {
    store,
    createContacts: jest.fn(() => Promise.resolve())
  }

  const originalWrapper = mount(<OriginalAddContactContainer {...props} state={initialData} />)

  it('AddContact render everythink correct', () => {
    expect(originalWrapper.exists('.add-contact')).toEqual(true);
  });

  it('AddContactContainer snapshot test', () => {
    const ContactsContainerShallow = mount(<AddContactContainer store={store} />);
    expect(ContactsContainerShallow).toMatchSnapshot();
  });

  it('AddContact dispatch is called', () => {
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).createContacts();
    expect(dispatch.mock.calls.length).toEqual(1);
  });

  it('AddContact btn is call function dispatch', () => {
    window.alert = () => {};
    originalWrapper.find('button').simulate('click');
    expect(props.createContacts.mock.calls[0][0].name).toEqual('Default');
  });

  it('Add-contact-container catch placeholder', () => {
    const errProps = {
      store,
      createContacts: jest.fn(() => Promise.reject())
    }
    console.log = () => {};
    const errorWrapper = mount(<OriginalAddContactContainer {...errProps} state={initialData} />);
    errorWrapper.find('button').simulate('click');
    expect(props.createContacts.mock.calls.length).toEqual(1);
  });

  it('Add-contact-container componentDidUpdate past a right id', () => {
    const initialContacts = {
      contacts: {}
    }
    const propsWrapper = mount(<OriginalAddContactContainer {...props} state={initialContacts} />);
    propsWrapper.setProps({state: initialData})
    const index = initialData.contacts.length -1;
    expect(propsWrapper.state().data.id).toEqual(initialData.contacts[index].id + 1)
  });

  it('Input and editContact(handler) for name work correct', () => {
    const inputDataType = originalWrapper.find('#name');
    const inputDataTypeConfig = { target: { value: 'NameCorrect', id: 'name' } };

    inputDataType.simulate('input', inputDataTypeConfig);

    expect(originalWrapper.state('data').name).toEqual(inputDataTypeConfig.target.value);
  });

  it('Input and editContact(handler) for lastname work correct', () => {
    const inputDataType = originalWrapper.find('#lastname');
    const inputDataTypeConfig = { target: { value: 'LastnameCorrect', id: 'lastname' } };

    inputDataType.simulate('input', inputDataTypeConfig);

    expect(originalWrapper.state('data').lastname).toEqual(inputDataTypeConfig.target.value);
  });

  it('Input and editContact(handler) for number work correct', () => {
    const inputDataType = originalWrapper.find('#number');
    const inputDataTypeConfig = { target: { value: 'NumberCorrect', id: 'number' } };

    inputDataType.simulate('input', inputDataTypeConfig);

    expect(originalWrapper.state('data').number).toEqual(inputDataTypeConfig.target.value);
  });

  it('Input and editContact(handler) for email work correct', () => {
    const inputDataType = originalWrapper.find('#email');
    const inputDataTypeConfig = { target: { value: 'EmailCorrect', id: 'email' } };

    inputDataType.simulate('input', inputDataTypeConfig);

    expect(originalWrapper.state('data').email).toEqual(inputDataTypeConfig.target.value);
  });

  it('Input and editContact(handler) for address work correct', () => {
    const inputDataType = originalWrapper.find('#address');
    const inputDataTypeConfig = { target: { value: 'AddressCorrect', id: 'address' } };

    inputDataType.simulate('input', inputDataTypeConfig);

    expect(originalWrapper.state('data').address).toEqual(inputDataTypeConfig.target.value);
  });

});