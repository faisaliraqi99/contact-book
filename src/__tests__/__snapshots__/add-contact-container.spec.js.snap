// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddContactContainer test group AddContactContainer snapshot test 1`] = `
<Connect(AddContactContainer)
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <AddContactContainer
    createContacts={[Function]}
    state={
      Object {
        "contacts": Array [],
        "isLoading": null,
        "selectedContact": Object {
          "address": "Select Contact",
          "email": "Select Contact",
          "lastname": "Select Contact",
          "name": "Select Contact",
          "number": "Select Contact",
        },
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <AddContact
      createContact={[Function]}
      editState={[Function]}
    >
      <div
        className="container"
      >
        <div
          className="add-contact"
        >
          <input
            className="contact-input"
            id="name"
            onInput={[Function]}
            placeholder="Name"
            type="text"
          />
          <input
            className="contact-input"
            id="lastname"
            onInput={[Function]}
            placeholder="Lastname"
            type="text"
          />
          <input
            className="contact-input"
            id="number"
            onInput={[Function]}
            placeholder="Number"
            type="text"
          />
          <input
            className="contact-input"
            id="email"
            onInput={[Function]}
            placeholder="Email"
            type="text"
          />
          <input
            className="contact-input"
            id="address"
            onInput={[Function]}
            placeholder="Address"
            type="text"
          />
          <button
            className="add-contact-btn"
            onClick={[Function]}
          >
            Create Contact
          </button>
        </div>
      </div>
    </AddContact>
  </AddContactContainer>
</Connect(AddContactContainer)>
`;
